### ISSUE TITLE
Support ruleset exclusion (eg. # noqa) in spec file 

### ISSUE THREAD

**Issue by ioggstream** [stoplightio/vscode-spectral#120]
**User story.**
As a {developer}, I would like to  {add skip directives in an OAS file}, so that I can {avoid modifying spectral.yml}.

**Is your feature request related to a problem?**
I struggle when I need to flag false positives during validation

**Describe the solution you'd like**
A "safe" proposal
```
openapi: 3.0.1
x-spectral-skip:
- rule-to-skip-1
- rule-to-skip-2
```

A "less safe" proposal

```
openapi: 3.0.1
x-spectral-rules:
  rule-1: false
  rule-2: hint
  rule-3: warn
```

**Additional context**
- the linter should just parse the document as yaml and get the `x-spectral-skip` field, then
  update the `ruleset` object, eg something like
```
skip_rules = Object.fromEntries(
  Object.entries(l).map( 
        k  => { [k[1], false] }
  )
)
ruleset={ ...ruleset, ...skip_rules}
```

**Comment by philsturgeon** [comment]
Could you talk me through the use case here a bit, when your OpenAPI is easier to change than your own config file? 

Even a distributed rule set can be configured  locally, so you should always be able to change your spectral config either just as easily as OpenAPI, or far easier.

**Comment by ioggstream** [comment]
> Could you talk me through the use case here a bit, when your OpenAPI is easier to change than your own config file? 

It is reasonable to change spectral.yml per-project, but

when exceptions are only related to single files Eg. because of false positives, it can be problematic to distribute for each file its ruleset counterpart.

Moreover as times goes by, this could lead to a proliferation of ruleset files with little differences.

The use case is then the same of  the noqa keyword used in other linters.

**Comment by philsturgeon** [comment]
Generally this is solved by publishing a distributable ruleset (e.g. https://github.com/digitalocean/openapi/blob/main/spectral/ruleset.yml) then your API will have a `.spectral.yml` that is skipping or excepting various rules that it is unable to conform to in the code repo its in with the openapi thats also in there. 

This is also basically how eslint works, with a config distributed via npm and a local config file. 

I don't think we'd want to add more ways to handle that, especially not changing OpenAPI documents to essentially configure spectral. 

Sorry if thats not the outcome you were looking for!
